torchtext/datasets/nli.py

```diff
@@ -161,3 +161,29 @@ def splits(cls, text_field, label_field, parse_field=None, genre_field=None,
                                            extra_fields=extra_fields,
                                            root=root, train=train,
                                            validation=validation, test=test)
+
+
+class XNLI(NLIDataset):
+    urls = ['http://www.nyu.edu/projects/bowman/xnli/XNLI-1.0.zip']
+    dirname = 'XNLI-1.0'
+    name = 'xnli'
+
+    @classmethod
+    def splits(cls, text_field, label_field, genre_field=None, language_field=None,
+               root='.data',
+               validation='xnli.dev.jsonl',
+               test='xnli.test.jsonl'):
+        extra_fields = {}
+        if genre_field is not None:
+            extra_fields["genre"] = ("genre", genre_field)
+        if language_field is not None:
+            extra_fields["language"] = ("language", language_field)
+
+        return super(XNLI, cls).splits(text_field, label_field,
+                                       extra_fields=extra_fields,
+                                       root=root, train=None,
+                                       validation=validation, test=test)
+
+    @classmethod
+    def iters(cls, *args, **kwargs):
+        raise NotImplementedError('XNLI dataset does not support iters')
```

Comment:
So why XNLI dataset doesn't support iters?

Reply:
Because of [this](https://github.com/pytorch/text/blob/master/torchtext/datasets/nli.py#L120) line onward. `NLIDataset` always assumes there is a training set, which is not the case for the XNLI dataset. I can edit the `NLIDataset` class to check if `train` is `None` and act accordingly?